code/client/munkilib/updatecheck.py

```diff
@@ -2751,16 +2760,19 @@ def download_icons(item_list):
                 munkicommon.display_error(
                     'Could not create %s' % icon_subdir)
                 continue
-        munkicommon.display_detail('Getting icon %s...', icon_name)
-        item_name = item.get('display_name') or item['name']
-        message = 'Getting icon for %s...' % item_name
-        try:
-            dummy_value = getResourceIfChangedAtomically(
-                icon_url, icon_path, message=message)
-        except fetch.MunkiDownloadError, err:
-            munkicommon.display_debug1(
-                'Could not retrieve icon %s from the server: %s',
-                icon_name, err)
+        if pkginfo_icon_hash != xattr_hash:
```

Comment:
More importantly:

Not what most _admin_ would want, and most importantly:

Would break existing sites (IOW, icons would no longer be retrieved, since no existing sites other than you are using pkginfo_icon_hash)